systemtests/test_end_to_end.py

```diff
@@ -0,0 +1,236 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Linux only!
+Tests that data can traverse through the autoreduction system successfully
+"""
+from __future__ import print_function
+import os
+import unittest
+import json
+import time
+import shutil
+
+
+from scripts.manual_operations import manual_remove as remove
+
+from utils import service_handling as external
+from utils.test_settings import MYSQL_SETTINGS, ACTIVEMQ_SETTINGS
+from utils.clients.database_client import DatabaseClient
+from utils.clients.queue_client import QueueClient
+from utils.data_archive.data_archive_creator import DataArchiveCreator
+from utils.data_archive.archive_explorer import ArchiveExplorer
+from utils.project.structure import get_project_root
+
+
+if os.name != 'nt':
+    class TestEndToEnd(unittest.TestCase):
+        """ Class to test pipelines in autoreduction"""
+
+        @classmethod
+        def setUpClass(cls):
+            # Start all services
+            external.start_activemq()
+            external.start_queue_processors()
+
+        def setUp(self):
+            """ Start all external services """
+            # Get all clients
+            self.database_client = DatabaseClient(MYSQL_SETTINGS)
+            self.database_client.connect()
+            self.queue_client = QueueClient(ACTIVEMQ_SETTINGS)
+            self.queue_client.connect()
+            # Create test archive and add data
+            self.data_archive_creator = DataArchiveCreator(os.path.join(get_project_root()))
```

Comment:
Should look for existing data archive in case the tests failed last time.

Reply:
Similar to above this should be handled more globally by a wrapper class. However, (also like above) this large change should be in a different issue. For the time being I will add a check for this to see if an archive already exists